### PR TITLE
Better tag rendering in intelligence view

### PR DESCRIPTION
**Pull request by tomchop**
This PR makes tag rendering a bit nicer in the intelligence view, prioritizing some tags given a pre-defined wieght and color class.

Additionally: 

* Constants such as `IOCTypes` were moved to their own file to avoid code duplication.
* Added search button for tags.
* Swapped column containing "copy to clipboard" / "search for IOC" buttons.
* Made trash icon have a pointer cursor while hovering.

With these settings:

```javascript
const tagMetadata = {
    malware: {
        weight: 100,
        class: 'danger',
    },
    suspicious: {
        weight: 50,
        class: 'warning',
    },
    legit: {
        weight: 10,
        class: 'success',
    },
    default: {
        weight: 0,
        class: 'info'
    }
}
```

image

### CODE REVIEW

File: timesketch/frontend/src/views/Intelligence.vue [hunk 1]
```diff
@@ -94,39 +105,41 @@ limitations under the License.
...
 
                   <b-table-column field="tags" label="Tags" v-slot="props">
                     <b-taglist>
-                      <b-tag v-for="tag in props.row.tags" v-bind:key="tag" type="is-info is-light">{{ tag }} </b-tag>
+                      <b-tag
+                        v-for="tag in getEnrichedTags(props.row.tags)"
+                        v-bind:key="tag.name"
+                        :type="`is-${tag.class} is-light`"
+                      >
+                        <router-link :to="{ name: 'Explore', query: generateOrElasticQuery(tagInfo[tag.name].iocs) }">
```
Comment (berggren) [inline timesketch/frontend/src/views/Intelligence.vue:119]: Please refactor to name this `generateOrOpenSearchQuery`

File: timesketch/frontend/src/views/Intelligence.vue [hunk 2]
```diff
@@ -298,9 +301,20 @@ export default {
...
+    enrichTag(tag) {
+      let tagInfo = { name: tag }
+      if (tagMetadata[tag]) {
+        return _.extend(tagInfo, tagMetadata[tag])
+      } else {
+        return _.extend(tagInfo, tagMetadata.default)
+      }
+    },
+    // TODO: Use filter chips instead
+    generateOrElasticQuery(valueList) {
```
Comment (berggren) [inline timesketch/frontend/src/views/Intelligence.vue:316]:
```suggestion
    generateOrOpenSearchQuery(valueList) {
```

File: timesketch/frontend/src/views/Intelligence.vue [hunk 3]
```diff
@@ -298,9 +301,20 @@ export default {
...
+      let query = valueList.map((v) => `"${v}"`).reduce((a, b) => `${a} OR ${b}`)
       return { q: query }
     },
     generateElasticQuery(value, field) {
```
Comment (berggren) [inline timesketch/frontend/src/views/Intelligence.vue:320]:
```suggestion
    generateOpenSearchQuery(value, field) {
```

**Review by berggren — CHANGES_REQUESTED** [review]
Just a small naming issue

**Comment by berggren** [comment]
PTAL @tomchop

**Comment by tomchop** [comment]
PTAL @ [4f1fca0](https://github.com/google/timesketch/pull/2123/commits/4f1fca0b6c7dfacf1d80ba6b8a3ecd09fb19397c) (add tag coloring and sorting to bottom left pane) and [d56ec5e](https://github.com/google/timesketch/pull/2123/commits/d56ec5ea19b46e050a8cb6f47779d1862da20da0) (refactor names)

**Review by berggren — APPROVED** [review]
LGTM

**Pull request merged**